pkg/provisioner/ironic/ironic.go

```diff
@@ -1289,6 +1289,15 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 }
 
 func (p *ironicProvisioner) setMaintenanceFlag(ironicNode *nodes.Node, value bool) (result provisioner.Result, err error) {
+	if ironicNode == nil {
+		p.log.Info("no node found, already deleted")
+		return result, nil
+	}
+	if ironicNode.Maintenance == value {
```

Comment:
This method isn't being called anywhere without checking for this first, so what's the problem?

Reply:
Seems so, as of current code, but using this check in local scope of this function ensures that this check will happen, avoids human error and avoids unneeded call to Ironic.

I should probably remove this checks from other places and leave only one here.

What do you think?

Reply:
I think that you won't be able to distinguish this case from others using the return value in the calling code and hence you will find there is no correct way to remove the current checks without adding another return value.